docs/DeveloperGuide.md

```diff
@@ -1,34 +1,956 @@
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[4.3.3.2. Design Considerations](#4332-design-considerations) <br>
+&nbsp;&nbsp;[4.4. Notebook Mode](#44-storage-neil) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;[4.4.1. Storage Format](#441-storage-format) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[4.4.1.1. TaskList](#4411-tasklist) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[4.4.1.2. Page](#4412-page) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[4.4.1.3. Section](#4413-section) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[4.4.1.4. Notebook](#4414-notebook) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[4.4.1.5. NotebookShelf](#4415-notebookshelf) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;[4.4.2. Implementation](#442-implementation) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[4.4.2.1. Saving the application state](#4421-saving-the-application-state) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[4.4.2.2. Reading the application state](#4422-reading-the-application-state) <br>
+&nbsp;&nbsp;[4.4. [Proposed] Find duplicates](#44-find-duplicates) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;[4.4.1 Proposed implementation](#441-proposed-implementation) <br>
+&nbsp;&nbsp;&nbsp;&nbsp;[4.4.2 Design considerations](#442-design-considerations) <br>
+[5. Documentation](#5-documentation) <br>
+&nbsp;&nbsp;[5.1. Setting up and maintaining the project website](#51-setting-up-and-maintaining-the-project-website) <br>
+&nbsp;&nbsp;[5.2. Style guidance](#52-style-guidance) <br>
+&nbsp;&nbsp;[5.3. Diagrams](#53-diagrams) <br>
+&nbsp;&nbsp;[5.4. Converting a document to the PDF Format](#54-converting-a-document-to-the-pdf-format) <br>
+[6. Testing](#6-testing) <br>
+&nbsp;&nbsp;[6.1. Running tests](#61-running-tests) <br>
+&nbsp;&nbsp;[6.2. Types of tests](#62-types-of-tests) <br>
+[Appendix A: Project Scope](#appendix-a-project-scope) <br>
+[Appendix B: User Stores](#appendix-b-user-stories) <br>
+[Appendix C: Use Cases](#appendix-c-use-cases) <br>
+[Appendix D: Non-Functional Requirements](#appendix-d-non-functional-requirements) <br>
+[Appendix E: Glossary](#appendix-e-glossary) <br>
+[Appendix F: Instructions for manual testing](#appendix-f-instructions-for-manual-testing) <br>
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## 1. Introduction (Neil)
 
+### 1.1. Welcome!
+Welcome, and thank you for choosing to help contribute to Zer0Note! Zer0Note is a command-line based note-taking and
+organisation application. It is designed to combine the features of graphical tools like OneNote and Notion, with
+the editing speed of applications like vim and emacs.
 
-## Product scope
-### Target user profile
+This document is written for developers intending to improve Zer0Note, by fixing bugs, or perhaps adding entirely new
+features. It explains how the project is set up, the architecture used, and the code style you should adopt when
+contributing code to the project.
 
-{Describe the target user profile}
+### 1.2. How to use this document
```

Comment:
I was a bit confused when reading this section. It kind of looks like one giant section and I had some trouble distinguishing one example from the next. Perhaps there could be a better way to separate/space out the examples to make it clearer?

Reply:
Thanks for the tip. I wrote this section, and looking at it I definitely see what you mean. We'll experiment with subheadings and `<hr>`s to try and make it more readable.

Reply:
There is some design or alignment issues on the ListCommand sequence diagram. Maybe you want to look into it.

Reply:
There is error in the Storage UML Diagram, I cant view it from my end. Maybe you want to look into it

Reply:
Yup, thank you! We will fix it.